lib/widgets/checkbox_widget.dart

```diff
@@ -0,0 +1,138 @@
+  List<Todo> taskList = [];
+
+  Future<List<Todo>> _getTask() async {
+    final _dataTasks =
+        Services.of(context).notesService.getTaskById(widget.note.id);
+    //
+
+    return _dataTasks;
+  }
+
+  @override
+  void initState() {
+    Future.microtask(
+      () => Provider.of<ProviderData>(context, listen: false)
+          .updateColor(widget.note.colorNote),
+    );
+    Provider.of<ProviderData>(context, listen: false).removeExit();
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<List<Todo>>(
+      future: _getTask(),
+      builder: (context, snaphot) {
+        final _tasks = snaphot.data ?? [];
+        //taskList = _tasks;
+        context.watch<ProviderData>().taskListProvider = _tasks;
+        if (snaphot.hasData) {
+          return ListView.builder(
+            itemCount: context.watch<ProviderData>().taskListProvider.length,
+            itemBuilder: (context, index) {
+              return GestureDetector(
+                onLongPress: () {
+                  Services.of(context)
+                      .notesService
+                      .deleteTask(_tasks[index].id);
+                  // ignore: no-empty-block
+                  setState(() {});
+                },
+                child: Row(
+                  children: [
+                    Expanded(
+                      child: CheckboxListTile(
+                        secondary: RemoveTaskWidget(
+                          task: _tasks[index].id,
+                          taskIndex: index,
+                        ),
+                        checkColor:
+                            Provider.of<ProviderData>(context).getTextColor,
+                        activeColor: colorPallete[widget.note.colorNote],
+                        title: Text(
+                          context
+                              .watch<ProviderData>()
+                              .taskListProvider[index]
+                              .task,
+                          style: TextStyle(
+                            fontSize: fontSizeK,
+                            decoration: context
+                                    .watch<ProviderData>()
+                                    .taskListProvider[index]
+                                    .check
+                                ? TextDecoration.lineThrough
+                                : TextDecoration.none,
+                            color: context
+                                    .watch<ProviderData>()
+                                    .taskListProvider[index]
+                                    .check
+                                ? context
+                                    .read<ProviderData>()
+                                    .getTextColor
+                                    .withOpacity(textOpacity)
+                                : Provider.of<ProviderData>(context)
+                                    .getTextColor,
```

Comment:
Introduce a few variables to make this more readable and reduce duplication.
```suggestion
                            decoration: taskIsChecked
                                ? TextDecoration.lineThrough
                                : TextDecoration.none,
                            color: taskIsChecked
                                ? checkedTextOpacity
                                : Provider.of<ProviderData>(context)
                                    .getTextColor,
```